features/294-login-without-specifying-login-id-key/api.md

```diff
@@ -105,62 +118,82 @@ skygear.auth.login(
 ### Sample use case: signup with custom loginID
 
 ```javascript
-// signup with a dict of loginIDs
 skygear.auth.signup({
-  role: 'example',
-  phone: '8912345',
-}, 'password');
+  "role-phone": {
+    role: 'example',
+    phone: '8912345',
+  },
+  username: 'exmaple',
+  email: 'example@example.com'
+}, 'password', {age: 18});
+// creates 3 loginIDs
+// 1. username: 'example',
+// 2. email: 'example@example.com'
+// 3. role-phone: { role: 'example , phone: '8912345' }
 skygear.auth.login(
-  'example', 'password'
+  {
+    role: 'example',
+    phone: '8912345',
+  }
 ).then((user) => {
   console.log(user);
   /*
+  LoginID-Source: role-phone,
   {
       user_id: <id>,
-      loginID_source: 'role'
-      metadata: {
-        // loginIDs
+      "role-phone": {
         role: 'example',
-        phone: '8912345'
+        phone: '8912345',
+      },
+      username: 'example',
+      email: 'example@example.com'
+      metadata: {
+        age: 18
       }
   }
   */
 });
+// or
 skygear.auth.login(
-  '8912345', 'password'
+  'example'
 ).then((user) => {
   console.log(user);
   /*
+  LoginID-Source: username,
   {
       user_id: <id>,
-      loginID_source: 'phone'
-      metadata: {
-        // loginIDs
+      "role-phone": {
```

Comment:
I think it is cleared to put all loginIDs under `loginIDs: {}` instead of flatten them on user object.